packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -104,8 +117,25 @@ class Doctor {
 
     bool allGood = true;
 
+    final Set<ValidatorCategory> finishedGroups = new Set<ValidatorCategory>();
     for (DoctorValidator validator in validators) {
-      final ValidationResult result = await validator.validate();
+      final ValidatorCategory currentCategory = validator.category;
+      ValidationResult result;
+
+      if (currentCategory.isGrouped) {
+        if (finishedGroups.contains(currentCategory)) {
```

Comment:
When could this happen? Should this be an assert instead?

Reply:
I meant to add to finishedGroups after handling a grouped category.

Reply:
What I meant was whether this is a natural occurrence or not. If the validators are grouped properly and each group executes once, why would we ever see a finished group? If this is valid case, please explain it as a comment above "continue". If it is not a valid case, please assert it (i.e. throw IllegalStateException if it happens).

Reply:
We're iterating over validators rather than categories. The first time we encounter a category, we skip ahead and round up all the output associated with that category. But we don't modify the validators list at all, so we'll continue to encounter that category (once for each validator in it).